Makefile

```diff
@@ -30,9 +30,10 @@ install:
 	# make sure /tmp in the snap is mode 1777
 	$(SUDO) chmod 1777 binary/boot/filesystem.dir/tmp
 	$(SUDO) mv binary/boot/filesystem.dir/* $(DESTDIR)/
-	# only copy the manifest file if we are in a launchpad buildd
+	# only copy the manifest and dpkg.yaml files if we are in a launchpad buildd
 	if [ -e /build/core ]; then \
 	  $(SUDO) mv livecd.ubuntu-core.manifest /build/core/core_16-$$(cat $(DESTDIR)/usr/lib/snapd/info|cut -f2 -d=|cut -f1 -d~|cut -b1-29)_$(DPKG_ARCH).manifest; \
+	  $(SUDO) cp usr/share/snappy/dpkg.yaml /build/core/core_16-$$(cat $(DESTDIR)/usr/lib/snapd/info|cut -f2 -d=|cut -f1 -d~|cut -b1-29)_$(DPKG_ARCH).dpkg.yaml; \
```

Comment:
It would be nice to create a variable containing all those `cut` pipes.

Reply:
@mardy Thanks for the review, I added a variable for the`cut` pipes and tested under LP Buildd itself that the snap builds successfully and produces the expected files (copied in the right places) with the correct version: https://launchpad.net/~ilasc/+snap/core-recipe/+build/1532608 and buildlog for details: https://launchpadlibrarian.net/559807750/buildlog_snap_ubuntu_xenial_amd64_core-recipe_BUILDING.txt.gz

Reply:
I actually meant something like this:

    TARGET_BASENAME=/build/core/core_16-$$(cat $(DESTDIR)/usr/lib/snapd/info|cut -f2 -d=|cut -f1 -d~|cut -b1-29)_$(DPKG_ARCH) \
    $(SUDO) mv livecd.ubuntu-core.manifest $(TARGET_BASENAME).manifest; \
    $(SUDO) cp /build/core/parts/livebuild/install/usr/share/snappy/dpkg.yaml $(TARGET_BASENAME).dpkg.yaml; \

(untested)

Sorry for the confusion! :-)

Reply:
@mardy Thanks for the suggestion Alberto! I tried something similar yesterday and various forms of it, it looks like any variable that contains the `info` file will result in failure (from what I could work out is evaluates before the file exists) - the build will be missing the needed `dpkg` and `manifest` files: https://launchpad.net/~ilasc/+snap/core-recipe/+build/1532705,  build log:  https://launchpadlibrarian.net/559822471/buildlog_snap_ubuntu_xenial_amd64_core-recipe_BUILDING.txt.gz
The only way I found to produce a successful build with the necessary files and correct suffix is the using a variable that doesn't attempt to contract that command any further than the `cut` pipes. 

Reply:
Ah, I think that we have to play a bit with `make` in order to properly expand the variable. This should work:
```bash
TARGET_BASENAME=/build/core/core_16-$$(cat $(DESTDIR)/usr/lib/snapd/info|cut -f2 -d=|cut -f1 -d~|cut -b1-29)_$(DPKG_ARCH) \
$(SUDO) mv livecd.ubuntu-core.manifest $${TARGET_BASENAME}.manifest; \
$(SUDO) cp /build/core/parts/livebuild/install/usr/share/snappy/dpkg.yaml $${TARGET_BASENAME}.dpkg.yaml; \
```

Reply:
@mardy Thanks for the snippet Alberto! I did try this approach too this week but it didn't work out either. I did however paste your snippet into the Production Core Snap to test it against the ral Buildd setup I have there, code here: https://git.launchpad.net/~ilasc/+git/core/tree/?h=cut-pipe-variables, the build that it produced missing the needed `dpkg` and `manifest` files: https://launchpad.net/~ilasc/+snap/core-recipe/+build/1533696 with buildlog: https://launchpadlibrarian.net/559958860/buildlog_snap_ubuntu_xenial_amd64_core-recipe_BUILDING.txt.gz.

I'm happy to keep trying out suggestions although I did make a genuine effort this week in producing a valid `Makefile` with a variable for the full path and the only way I found to produce a successful build with the necessary files and correct suffix is the using a variable that doesn't attempt to contract that command any further than the `cut` pipes. 

Let me know what you think, thanks!

Reply:
Thanks for your patience! Indeed make has a weird way of handling variables... It's strange because I'm using the `$${}` in other debian/rules files, and it appears to be working there. Can you please try different options? Like, using only `$$TARGET_BASENAME` (without braces), then with a single `$`, then `$$(...)`... there must be a working syntax! :-)

Ah, wait, as first thing, try removing the `_` and name the variable `TARGETBASENAME`, just in case `make` doesn't like that...

Reply:
Thanks @mardy, agreed on this being strange! :-) I will start with all those different options now and I'll share later links to all outputs generated.

Reply:
Just for your information, @ilasc, I'm not a maintainer of this project; I only happened to see the notification of this change, and thought that the code could be improved. But I absolutely not my intention to block you or make you spend a lot of time on this. :-)

If @mvo is fine with your solution, so I am :-)

Reply:
@mardy I appreciate the message, I fully agreed a variable improve the code, unfortunately none of my attempts worked, while I did find a few combinations that worked for `make` locally, as soon as I put them through Buildd's build process of this snap they all failed... 

I've also put your suggestions above through the Launchpad Buildd setup with no luck:

1: Commit `Remove _ from TARGET_BASENAME`: https://git.launchpad.net/~ilasc/+git/core/commit/?id=fc1cb9a8201e43128aea3545a4791a02ad16a188 with Build: https://launchpad.net/~ilasc/+snap/core-recipe/+build/1533747 and Buildlog: https://launchpadlibrarian.net/559967584/buildlog_snap_ubuntu_xenial_amd64_core-recipe_BUILDING.txt.gz
2: Commit `Remove {} around TARGETBASENAME`: https://git.launchpad.net/~ilasc/+git/core/commit/?id=0e5e7dc92c44d9fdea5bd7ba004e120c4b142266 with Build: https://launchpad.net/~ilasc/+snap/core-recipe/+build/1533756 Buildlog: https://launchpadlibrarian.net/559971405/buildlog_snap_ubuntu_xenial_amd64_core-recipe_BUILDING.txt.gz
3: Commit `Use single $ for TARGETBASENAME`: https://git.launchpad.net/~ilasc/+git/core/commit/?id=f8b4af0b717a0c54588d0d4a5ce73c853d7adb14, Build: https://launchpad.net/~ilasc/+snap/core-recipe/+build/1533770, Buildlog: https://launchpadlibrarian.net/559980319/buildlog_snap_ubuntu_xenial_amd64_core-recipe_BUILDING.txt.gz

We'll see what @mvo has to say and thanks for your input Alberto!!

Reply:
Mmm... the last one is the most encouraging! Last attempt: `${TARGETBASENAME}` :-)

Reply:
@mardy I had already tried that one last week already unfortunately with the same results, but I've put it through Buildd again this morning to be able to share with you: Commit: https://git.launchpad.net/~ilasc/+git/core/commit/?id=da3eaf57d0c4b9b343e08192bc89735098809ced, Build without `manifest` and `dpkg` files:  https://launchpad.net/~ilasc/+snap/core-recipe/+build/1536824 and Buildlog: https://launchpadlibrarian.net/560367296/buildlog_snap_ubuntu_xenial_amd64_core-recipe_BUILDING.txt.gz

Reply:
Sorry for joining so late - I looked at your build number (2): `$$TARGETBASENAME` (which really ought to work) and there is a successful buildlog and also a resulting snap that has dpkg.yaml - is the buildlog misleading me?

Fwiw, I think it's okay to move forward without this but at the same time my experience is that things like this that are not fully understood usually come back and haunt us :) So I would really like to at least poke a bit at it to see where we miss something.

Reply:
@mvo5 Hi Michael,

For us to call a core snap build successful the build needs to produce this output: a `.snap` file, a `.manifest` file and a `dpkg.yaml` file, example of a successful build produced by the code proposed in this PR: https://launchpad.net/~ilasc/+snap/core-recipe/+build/1532608

Build build number (2): $$TARGETBASENAME is not a successful build as it is missing both the `manifest` file and `dpkg.yaml` file.

What is proposed in this PR is producing a successful build (https://launchpad.net/~ilasc/+snap/core-recipe/+build/1532608) and I maintain the statement in my early comment on the PR: it is pretty clear we cannot contract that path variable further than the code in this PR does as any variable that contains the `info` file will result in failure (it evaluates before the file exists) due to the way the snap build works under Buildd (as can be seen from the list of attempts made here with these builds: https://launchpad.net/~ilasc/+snap/core-recipe).

The goal of this PR is to copy alongside the `manifest` file the new `dpkg.yaml` which is does successfully. The discussion spawned here is around refactoring existing code that was already working successfully for moving the `manifest` file to make it more readable. I agree it's okay to move forward without refactoring any further the way we build that path for the files to be copied / moved to. 

Reply:
Let's land it and do a followup then.